script.js

```diff
@@ -1,4 +1,400 @@
+let noOfPlayers = 0;
+let playerProfiles = [];
+let dealerCards = [];
+let deck = [];
+let activePlayer = 0;
+let gameMode = 'start';
+let roundsPlayed = 0;
+
+const inputEl = document.querySelector('#input-field');
+const newGameBtn = document.querySelector('.new-game');
+const outputEl = document.querySelector('#output-div');
+const scoreboardEl = document.querySelector('#scoreboard-data');
+
+/* TODO
+add DOM elements for:
+1. New game button (done)
+2. Track wins of each player (done)
+3. Reshuffle deck when current deck is empty
```

Comment:
The technique you can use to reshuffle deck when current deck is empty, in pseudocode:
```
// 1. deckGenerator creates and shuffles deck created.
// 2. The deck is created once at line 384.
// 3. You will want to duplicate this deck and store it in a variable. The duplicated deck will be used to play the game>
// 4. Whenever you know the deck is about to run out (before anyone hits, or before cards are drawn for each player and dealer), duplicate the deck again (perhaps also reshuffle it) then re-assign it to the variable created in step (3)
```

We do not really emphasise deck reshuffles at Basics, because it requires an understanding of references vs. values, and the differences between (functions, arrays, and objects) and (other primitive values) in that situation.

This will be covered in Bootcamp though, and you should be expected to embark on this starting from Bootcamp Project 1. These are notes on [how you can make (shallow) copies of arrays](https://bootcamp.rocketacademy.co/0-0-language-and-tooling/0.2-es6/0.2.3-es6-destructuring-spread-operators#example-make-shallow-copy-of-array), and [the rationale for it](https://bootcamp.rocketacademy.co/0-0-language-and-tooling/0.2-es6/0.2.4-reference-vs.-value), if you want to read ahead.